test/parallel/test-stream-writable-finish-destroyed.js

```diff
@@ -0,0 +1,33 @@
+'use strict';
+
+const common = require('../common');
+const { Writable } = require('stream');
+
+{
+  const w = new Writable({
+    write(chunk, encoding, cb) {
+      w.on('close', () => {
+        cb();
+      });
+    }
+  });
+
+  w.end('asd');
+  w.destroy();
+  w.on('finish', common.mustNotCall());
```

Comment:
Can you please add a `once` handler to check if this is emitted? It _should_ in this case.

Reply:
What do you mean? `'finish'` should not be emitted in this case.

Reply:
then let's move this before .end(), so it's clear.

Reply:
moved